src/pages/YourPeep/YourOffChainPeep.tsx

```diff
@@ -0,0 +1,145 @@
+import { useContext, useEffect, useState } from 'react'
+import { MetadataContext } from '../../context/Metadata/MetadataContext'
+import { DEFAULT_PEEP, getTrait } from '../../interface/metadata'
+import useStyles from './YourOffChainPeep.styles'
+import { ProfileContext } from '../../context/Profile/ProfileContext'
+import doFetch from '../../utils/doFetch'
+import { host } from '../../config/api'
+import { Link, useLocation } from 'react-router-dom'
+import { HomeRoute } from '../routes'
+import { greetings } from '../../config/text'
+import Copy from '../../components/Button/Copy'
+
+interface propState {
+	uri: string
+	isUpdate: boolean
+}
+
+const YourPeep = () => {
+	const classes = useStyles()
+	const { metadata, setMetadata } = useContext(MetadataContext)
+	const { profile, setProfile } = useContext(ProfileContext)
+	const [yourPeepImage, setYourPeepImage] = useState('')
+	const [milkyGreeting, setMilkyGreeting] = useState('Aloha')
+	const [jonoGreeting, setJonoGreeting] = useState('Ciao')
+	const [grannyGreeting, setGrannyGreeting] = useState('Hello')
+	const [officerGreeting, setOfficerGreeting] = useState('Bonjour!')
+
+	if (
+		!metadata ||
+		!setMetadata ||
+		!profile ||
+		!setProfile ||
+		!useLocation().state
+	) {
+		return <></>
+	}
+
+	const getRandomGreeting = () => {
+		const i = Math.floor(Math.random() * greetings.length)
+		return greetings[i]
+	}
+	const { uri } = useLocation().state as propState
```

Comment:
Can we avoid `as propState`? It's a smell that types are wrong elsewhere.

Reply:
I don't think we can set the type of location state, for now I have made a new interface and now using it on both pages.